pkg/grizzly/parsing.go

```diff
@@ -18,13 +20,50 @@ import (
 )
 
 func Parse(registry Registry, opts GrizzlyOpts) (Resources, error) {
-	if strings.HasSuffix(*opts.ResourceFile, ".yaml") ||
-		strings.HasSuffix(*opts.ResourceFile, ".yml") {
-		return ParseYAML(registry, *opts.ResourceFile, opts)
-	} else if strings.HasSuffix(*opts.ResourceFile, ".jsonnet") ||
-		strings.HasSuffix(*opts.ResourceFile, ".libsonnet") ||
-		strings.HasSuffix(*opts.ResourceFile, ".json") {
-		return ParseJsonnet(registry, *opts.ResourceFile, opts)
+	if opts.ResourceFile != nil && opts.Directory != nil {
+		return nil, fmt.Errorf("Only specify a directory or a resource file, not both")
+	}
+	if opts.ResourceFile != nil {
+		return ParseFile(registry, opts, *opts.ResourceFile)
+	}
+	if opts.Directory != nil {
+		resources := Resources{}
+		files, err := FindResourceFiles(registry, opts, *opts.Directory)
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			r, err := ParseFile(registry, opts, file)
+			if err != nil {
+				return nil, err
+			}
+			resources = append(resources, r...)
+		}
+		return resources, nil
+	}
+	return nil, fmt.Errorf("Must specify either a resource file or a directory")
+}
+
+func FindResourceFiles(registry Registry, opts GrizzlyOpts, dir string) ([]string, error) {
+	files := []string{}
+	for _, handler := range registry.Handlers {
+		handlerFiles, err := handler.FindResourceFiles(dir)
+		if err != nil {
+			return nil, err
+		}
+		files = append(files, handlerFiles...)
+	}
+	return files, nil
+}
+
+func ParseFile(registry Registry, opts GrizzlyOpts, resourceFile string) (Resources, error) {
+	if strings.HasSuffix(resourceFile, ".yaml") ||
+		strings.HasSuffix(resourceFile, ".yml") {
+		return ParseYAML(registry, resourceFile, opts)
+	} else if strings.HasSuffix(resourceFile, ".jsonnet") ||
+		strings.HasSuffix(resourceFile, ".libsonnet") ||
+		strings.HasSuffix(resourceFile, ".json") {
+		return ParseJsonnet(registry, resourceFile, opts)
 	} else {
 		return nil, fmt.Errorf("Either a config file or a resource file is required")
 	}
```

Comment:
This screams for a switch! Also there is `filepath.Ext` ;)

```suggestion
func ParseFile(registry Registry, opts GrizzlyOpts, resourceFile string) (Resources, error) {
	switch filepath.Ext(resourceFile) {
	case ".yml", ".yaml":
		return ParseYAML(registry, resourceFile, opts)
	case ".jsonnet":
		return ParseJsonnet(registry, resourceFile, opts)
	}
	
	return nil, fmt.Errorf("Either a config file or a resource file is required")
```